web/api/web_api_v1.c

```diff
@@ -285,12 +286,13 @@ inline int web_client_api_request_v1_alarm_log(RRDHOST *host, struct web_client
         if(!name || !*name) continue;
         if(!value || !*value) continue;
 
-        if(!strcmp(name, "after")) after = (uint32_t)strtoul(value, NULL, 0);
+        if      (!strcmp(name, "after")) after = (uint32_t)strtoul(value, NULL, 0);
```

Comment:
Hello,

I can see in this line a tab instead a simple space between `if` and the parenthesis, but this is not what called my attention, I saw that the base was set to 0, and I will suggest we use 10, because the `0` can mean different things according the official documentation:

```
 The strtoul() function converts the initial part of the string in nptr to an unsigned long value according to the given base, which must be between 2 and 36 inclusive, or be the special value 0.

       The  string  may  begin  with an arbitrary amount of white space (as determined by isspace(3)) followed by a single optional '+' or '-' sign.  If base is zero or 16, the string may then include a "0x" prefix, and the number
       will be read in base 16; otherwise, a zero base is taken as 10 (decimal) unless the next character is '0', in which case it is taken as 8 (octal).

       The remainder of the string is converted to an unsigned long value in the obvious manner, stopping at the first character which is not a valid digit in the given base.  (In bases above 10, the letter 'A' in either uppercase
       or lowercase represents 10, 'B' represents 11, and so forth, with 'Z' representing 35.)
```

When I accessed the URL on browser to get a specific timestamp, I selected the following value:

![Screenshot_20210324_125853](https://user-images.githubusercontent.com/49162938/112314731-43cc0700-8ca1-11eb-835e-64e86718c90b.png)


But, when I ran another query I got a clean request. After to identify this problem I ran the following commands:

```bash
bash-5.1$ curl -k "https://localhost:19999/api/v1/alarm_log?chart=system.cpu" | wc -l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  111k  100  111k    0     0  13.6M      0 --:--:-- --:--:-- --:--:-- 13.6M
4626
bash-5.1$ curl -k "https://localhost:19999/api/v1/alarm_log?chart=system.cpu&after=1616590251" 
[
]
bash-5.1$ curl -k "https://localhost:19999/api/v1/data?chart=system.cpu"  | wc -l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 39257  100 39257    0     0  9584k      0 --:--:-- --:--:-- --:--:-- 12.4M
605
bash-5.1$ curl -k "https://localhost:19999/api/v1/data?chart=system.cpu&after=1616590535"  | wc -l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9072  100  9072    0     0  4429k      0 --:--:-- --:--:-- --:--:-- 4429k
144
```

As you can see it is not working as expected, please, compare with the `/api/v1/data`, because it looks like we are missing something here.

Reply:
I hadn't noticed that but it seems to work as expected. It just means that the `after` can also be in hexadecimal or octal. So the following are identical (in hex, decimal and octal)

`http://127.0.0.1:19999/api/v1/alarm_log?after=0x5F4DF281`

and

`http://127.0.0.1:19999/api/v1/alarm_log?after=1598943873`

and 

`http://127.0.0.1:19999/api/v1/alarm_log?after=013723371201`

Note that the `after` is applied to the unique_id